Class/Practise 5.1/src/MegaDataPool.cpp

```diff
@@ -0,0 +1,42 @@
+#include "MegaDataPool.hpp"
+#include <stdexcept>
+
+MegaDataPool::MegaDataPool(size_t poolSize)
+    : poolSize(poolSize), pool(poolSize), used(poolSize, false) {
+    for (size_t i = 0; i < poolSize; ++i) {
+        pool[i] = std::make_shared<MegaData>();
+    }
+}
+
+std::shared_ptr<MegaData> MegaDataPool::acquire() {
```

Comment:
Here we have an interesting question  - what will happen if someone will copy this shared pointer. Should we rely on user to follow the API or should we use some design solution to guarantee that. There are different ways to reach the goal and  i hope that it may be an interesting topic for investigation and reading.

Reply:
Improved by returning a shared pointer from acquire ensures that when it goes out of scope, the release function is called automatically. This will prevent race conditions where multiple threads might try to acquire or release objects simultaneously.